shotover-proxy/tests/helpers/cassandra.rs

```diff
@@ -1,36 +1,234 @@
+                }
+            }
+            CassandraDriver::CdrsTokio => {
+                let user = "cassandra";
+                let password = "cassandra";
+                let auth = StaticPasswordAuthenticatorProvider::new(&user, &password);
+
+                let node_addresses = contact_points
+                    .split(',')
+                    .map(|contact_point| NodeAddress::from(format!("{contact_point}:{port}")))
+                    .collect::<Vec<NodeAddress>>();
+
+                let config = NodeTcpConfigBuilder::new()
+                    .with_contact_points(node_addresses)
+                    .with_authenticator_provider(Arc::new(auth))
+                    .build()
+                    .await
+                    .unwrap();
+
+                let session =
+                    TcpSessionBuilder::new(RoundRobinLoadBalancingStrategy::new(), config)
+                        .build()
+                        .unwrap();
+                CassandraConnection::CdrsTokio {
+                    session,
+                    schema_awaiter: None,
+                }
+            }
+        }
+    }
+
+    #[allow(unused)]
+    pub fn as_cdrs(&self) -> &CdrsTokioSessionInstance {
+        match self {
+            Self::CdrsTokio { session, .. } => session,
+            _ => panic!("Not CdrsTokio"),
+        }
+    }
+
+    #[cfg(feature = "cassandra-cpp-driver-tests")]
+    #[allow(unused)]
+    pub fn as_datastax(&self) -> &DatastaxSession {
+        match self {
+            Self::Datastax { session, .. } => session,
+            _ => panic!("Not Datastax"),
+        }
+    }
+
+    #[allow(unused)]
+    pub fn new_tls(
+        contact_points: &str,
+        port: u16,
+        ca_cert_path: &str,
+        driver: CassandraDriver,
+    ) -> Self {
+        match driver {
+            #[cfg(feature = "cassandra-cpp-driver-tests")]
+            CassandraDriver::Datastax => {
+                let ca_cert = read_to_string(ca_cert_path).unwrap();
+                let mut ssl = Ssl::default();
+                Ssl::add_trusted_cert(&mut ssl, &ca_cert).unwrap();
+
+                for contact_point in contact_points.split(',') {
+                    test_helpers::wait_for_socket_to_open(contact_point, port);
+                }
+
+                let mut cluster = Cluster::default();
+                cluster.set_credentials("cassandra", "cassandra").unwrap();
+                cluster.set_contact_points(contact_points).unwrap();
+                cluster.set_port(port).ok();
+                cluster.set_load_balance_round_robin();
+                cluster.set_ssl(&mut ssl);
+
+                CassandraConnection::Datastax {
+                    session: cluster.connect().unwrap(),
+                    schema_awaiter: None,
+                }
+            }
+            // TODO actually implement TLS for cdrs-tokio
+            CassandraDriver::CdrsTokio => {
+                let user = "cassandra";
```

Comment:
maybe just make this branch into a todo!() instead of a bunch of code that doesnt work.